streaming/src/main/scala/org/apache/spark/streaming/scheduler/ReceiverTracker.scala

```diff
@@ -484,7 +528,10 @@ class ReceiverTracker(ssc: StreamingContext, skipReceiverLaunch: Boolean = false
         new SerializableConfiguration(ssc.sparkContext.hadoopConfiguration)
 
       // Function to start the receiver on the worker node
-      val startReceiverFunc = new StartReceiverFunc(checkpointDirOption, serializableHadoopConf)
+      val startReceiverFunc: Iterator[Receiver[_]] => Unit =
+        (i: Iterator[Receiver[_]]) => {
+          new StartReceiverFunc(checkpointDirOption, serializableHadoopConf)(i)
```

Comment:
You did not fix the `private class StartReceiverFunc` to `private[scheduler] class StartReceiverFunc`. Alternatively, it may be better to make this a private inner class of ReceiverTracker, rather than polluting the top level class list.